packages/lyra/src/prefix-tree/trie.ts

```diff
@@ -63,16 +63,35 @@ export class Trie {
       }
     }
 
+    // avoid the recursive call if the word is the same as the term
+    const [word, docIDs] = node.getWord();
+
+    if (exact && word === term) {
+      output[word] = new Set();
+
+      handleOutput(word, docIDs, output);
+
+      return output;
+    }
+
     findAllWords(node, output);
 
+    function handleOutput(
+      word: string,
+      docIDs: Set<string>,
+      _output: FindResult
+    ): void {
+      if (docIDs.size) {
+        for (const doc of docIDs) {
+          _output[word] && _output[word].add(doc);
```

Comment:
Heyy @mateonunez 👋🏽
I'm wondering if changing this in

```javascript
const val = _output[word];
if (docIDs.size && val) {
 for (const doc of docIDs) {
  val.add(doc);
 }
}
```
would prevent potential overhead on falsy `_output[word]`. 
- It seems that `_output[word]` is not changing (will stop accessing _output all the time).
- therefore we could enter the loop only if `_output[word]` is truthy. 

> Obviously the overhead/perfs depend on the size of the Set
<hr>
Anyway, thanks in advance for your feedback, eventually 😄

Reply:
I think the PR needs to make other logic, I will close this one trying to solve it definitively.